pandas/core/frame.py

```diff
@@ -5070,14 +5070,38 @@ def notna(self) -> DataFrame:
     def notnull(self) -> DataFrame:
         return ~self.isna()
 
+    @overload
+    # https://github.com/python/mypy/issues/6580
+    # Overloaded function signatures 1 and 2 overlap with incompatible return types
+    def dropna(  # type: ignore[misc]
+        self,
+        axis: Axis = ...,
+        how: str = ...,
+        thresh: Optional[int] = ...,
+        subset: Optional[Union[Hashable, Sequence[Hashable]]] = ...,
+        inplace: Literal[False] = ...,
+    ) -> DataFrame:
+        ...
+
+    @overload
+    def dropna(
+        self,
+        axis: Axis = ...,
+        how: str = ...,
+        thresh: Optional[int] = ...,
+        subset: Optional[Union[Hashable, Sequence[Hashable]]] = ...,
```

Comment:
we now have an `IndexLabel` alias in pandas._typing that could be used for subset.

```suggestion
        subset: Optional[IndexLabel] = ...,
```